### ISSUE TITLE
Still maintained?

### ISSUE THREAD

**Issue by ollydixon** [rikulo/socket.io-client-dart#276]
Is this library still maintained?

I see many of the releases are many months ago. Still no support for V4 unless you use the Beta [2.0.0-beta.4-nullsafety.0](https://pub.dev/packages/socket_io_client/versions/2.0.0-beta.4-nullsafety.0)

If it's not maintained can we get a deprecated flag on it so it pushes others to make alternative libraries :-)

**Comment by RukshanJS** [comment]
> Is this library still maintained?
> 
> I see many of the releases are many months ago. Still no support for V4 unless you use the Beta [2.0.0-beta.4-nullsafety.0](https://pub.dev/packages/socket_io_client/versions/2.0.0-beta.4-nullsafety.0)
> 
> If it's not maintained can we get a deprecated flag on it so it pushes others to make alternative libraries :-)

I agree. Although the `2.0.0-beta.4-nullsafety.0` version works, there are some nuances which cause errors for example issues during reconnecting the socket etc.

AFAIK one of the main issues with websockets is that the server and client sides should be compatible with each other. So currently the only working version (for me atleast) is this beta one. Would really love a new and maintained version.

**Comment by ollydixon** [comment]
@RukshanJS it's been very solid for us, on all platforms (Web, Android, iOS, MacOS, and Windows) using it on production feels a bit risky though.

**Comment by ollydixon** [comment]
I find that if you call disconnect, destroy or dispose it never works again, instead we just do this

`_socket?.clearListeners();`

Then remake it.

**Comment by huydq88** [comment]
I'm starting to worry about Flutter's retention in the future. And over the past 14 months, the Socket Io Client Beta version has not made any progress.

**Comment by ollydixon** [comment]
@huydq88 nah Flutter is booming, just this library needs some love

**Comment by RukshanJS** [comment]
> I find that if you call disconnect, destroy or dispose it never works again, instead we just do this
> 
> `_socket?.clearListeners();`
> 
> Then remake it.

Just wanted to thank you for this. This certainly helped clear out the excess listeners but still the reconnecting thing is not working as expected on a physical device. It just throws the `SocketException` and does not reconnect (but works fine on emulator).

Since this thread is for another discussion I'll try to debug and post a separate issue if this persists.

**Comment by ollydixon** [comment]
@jumperchen any thoughts?

**Comment by RukshanJS** [comment]
> @RukshanJS we're using it on many phyiscal devices, Android, iOS, MacOS Apps, Windows Apps, Web (lots of browses) and it reconnects just fine, perhaps it's a serverside issue?

😃 is that so? My backend is NestJS and using the latest versions of the packages @nestjs/platform-socket.io@8.4.7 and @nestjs/websockets@8.4.7. I think they use socket.io@4.5.1 under the hood. I wonder what causes it to throw that `SocketException` only on the physical device. Tested several times to make sure as well. I'll update here for sure if I make any progress. Appreciate your support.

**Comment by ollydixon** [comment]
I had issues with some of the socket packages that were a bridge to socket. I was using a apollo GraphQL API, scrapped the built in adaptor and just raw socket-io with a redis adaptor for scalability.

**Comment by RukshanJS** [comment]
> @RukshanJS I had issues with some of the socket packages that were a bridge to socket-io. I was using a Apollo GraphQL, scrapped their built in adaptor and used the raw socket-io with a redis adaptor for scalability.

Oh maybe I should give it a try. Thanks for the tip 😃

**Comment by ollydixon** [comment]
@rmControls could just use web-sockets, sure it requires a little more code and re-connection logic but it's possible and you get more control over bugs and connectivity.

**Comment by RukshanJS** [comment]
@ollydixon My socket reconnection issue is now fixed. I did some code cleaning and that must've fixed some listeners or instances of the socket (have to check it more). Now the reconnection is working on physical device as well as expected. So... it seems the package is ok like you said. Cheers!

**Comment by rmControls** [comment]
@RukshanJS if possible, can you please share the sample code which is working for you??

**Comment by neckaros** [comment]
I have the same issue you have i think: if server disconnect because of a restart. After reconnect (i get the ondisconnect followed by a onconnect event) all other listener are not called.

**Comment by GabrielBB** [comment]
Maybe the author is waiting for PRs and not just Issues :D ?

**Comment by GabrielBB** [comment]
@jumperchen just released a new version: https://github.com/rikulo/socket.io-client-dart/commit/e629e191657f482d155ee1574de5ca3f920b9d17

**Comment by ollydixon** [comment]
@GabrielBB great thanks. 

We really need union types 😂🙈 dynamic is dangerous.

<img width="554" alt="Screenshot 2022-07-14 at 22 15 53" src="https://user-images.githubusercontent.com/6940726/179075696-62865bab-a36a-423b-9ac7-18d2926c681b.png">

**Comment by RukshanJS** [comment]
> @RukshanJS if possible, can you please share the sample code which is working for you??

Hi, really sorry couldn't get back to you earlier. I have hosted the full source code (with NestJS backend + Flutter frontend) which was a tutorial project that I did.

In addition, I have written a complete tutorial that worked for me using this package. It is a three part series but if you want only the Flutter implementation, you can read the part 3 - https://dev.to/rukshanjs/part-33-how-to-create-a-server-side-timer-using-websockets-with-socketio-nestjs-and-flutter-10am or watch the part 3 on my YouTube channel https://youtu.be/kx7LQNNmTX4. 

Very probable that you have found a working code by now... thought to share anyways..

**Comment by rmControls** [comment]
@RukshanJS thank you... that still helps....
